crates/turbopack-dev-server/src/source/mod.rs

```diff
@@ -311,6 +374,28 @@ impl ContentSourceDataVary {
         ContentSourceDataFilter::extend_options(&mut self.query, &other.query);
         ContentSourceDataFilter::extend_options(&mut self.headers, &other.headers);
     }
+
+    pub fn fulfills(&self, other: &ContentSourceDataVary) -> bool {
+        if other.method && !self.method {
+            return false;
+        }
+        if other.url && !self.url {
+            return false;
+        }
+        if other.body && !self.body {
+            return false;
+        }
+        if other.cache_buster && !self.cache_buster {
+            return false;
+        }
+        if !ContentSourceDataFilter::fulfills(&self.query, &other.query) {
+            return false;
+        }
+        if !ContentSourceDataFilter::fulfills(&self.headers, &other.headers) {
+            return false;
+        }
+        true
```

Comment:
This could easily break if we were to add new extensions to `ContentSourceDataVary`.

I would argue that either:
* `fulfills` should be a trait that is automatically derived; or
* we use destructuring assignment in this function to not forget any fields.

```rust
let ContentSourceDataVary { method: other_method, /* etc., but no `..` */ } = &other;
```

But this clashes with `placeholder_for_future_extensions`.